llvm/lib/Target/AMDGPU/SIFrameLowering.cpp

```diff
@@ -1662,6 +1714,21 @@ void SIFrameLowering::determineCalleeSaves(MachineFunction &MF,
   if (MFI->isEntryFunction())
     return;
 
+  if (MFI->isWholeWaveFunction()) {
+    // In practice, all the VGPRs are WWM registers, and we will need to save at
+    // least their inactive lanes. Add them to WWMReservedRegs.
```

Comment:
Are we considering all VGPRs to be WWM regs in this calling convention?

Reply:
For the purposes of prolog epilog insertion, yes, since we need to preserve all the inactive lanes.